### DISCUSSION TITLE
PEBKAC :: cannot save lists

### DISCUSSION THREAD

**Discussion by sinkr** [ThePrimeagen/harpoon#585] [Q&A]
Definitely my fault, but nothing I do will make my lists persist sessions.  I've tried numerous iterations of settings in the .lua file to no avail.  Does someone have an end-to-end LazyVim config that just works?  Apparently I'm too lame to take the snippets and adapt them.

**Comment by maxrzaw** [comment]
I am not exactly sure what you mean by end-to-end, but adding this file to the starter LazyVim config at `lua/plugins/harpoon.lua` works.
```lua
-- lua/plugins/harpoon.lua
return {
    "ThePrimeagen/harpoon",
    branch = "harpoon2",
    dependencies = { "nvim-lua/plenary.nvim" },
    config = function()
        local harpoon = require("harpoon")

        -- REQUIRED
        harpoon:setup()
        -- REQUIRED

        vim.keymap.set("n", "<leader>a", function() harpoon:list():add() end)
        vim.keymap.set("n", "<C-e>", function() harpoon.ui:toggle_quick_menu(harpoon:list()) end)

        vim.keymap.set("n", "<C-h>", function() harpoon:list():select(1) end)
        vim.keymap.set("n", "<C-t>", function() harpoon:list():select(2) end)
        vim.keymap.set("n", "<C-n>", function() harpoon:list():select(3) end)
        vim.keymap.set("n", "<C-s>", function() harpoon:list():select(4) end)

        -- Toggle previous & next buffers stored within Harpoon list
        vim.keymap.set("n", "<C-S-P>", function() harpoon:list():prev() end)
        vim.keymap.set("n", "<C-S-N>", function() harpoon:list():next() end)
    end,
}
```

**Reply by OriLiMu** [reply]
> I am not exactly sure what you mean by end-to-end, but adding this file to the starter LazyVim config at `lua/plugins/harpoon.lua` works.
> 
> ```lua
> -- lua/plugins/harpoon.lua
> return {
>     "ThePrimeagen/harpoon",
>     branch = "harpoon2",
>     dependencies = { "nvim-lua/plenary.nvim" },
>     config = function()
>         local harpoon = require("harpoon")
> 
>         -- REQUIRED
>         harpoon:setup()
>         -- REQUIRED
> 
>         vim.keymap.set("n", "<leader>a", function() harpoon:list():add() end)
>         vim.keymap.set("n", "<C-e>", function() harpoon.ui:toggle_quick_menu(harpoon:list()) end)
> 
>         vim.keymap.set("n", "<C-h>", function() harpoon:list():select(1) end)
>         vim.keymap.set("n", "<C-t>", function() harpoon:list():select(2) end)
>         vim.keymap.set("n", "<C-n>", function() harpoon:list():select(3) end)
>         vim.keymap.set("n", "<C-s>", function() harpoon:list():select(4) end)
> 
>         -- Toggle previous & next buffers stored within Harpoon list
>         vim.keymap.set("n", "<C-S-P>", function() harpoon:list():prev() end)
>         vim.keymap.set("n", "<C-S-N>", function() harpoon:list():next() end)
>     end,
> }
> ```

save config as you said the harpoon file list is empty after restart nvim. using lazy vim how to fix it?

**Comment by sinkr** [comment]
Hi @maxrzaw, what you posted was whtat I meant by end-to-end (not snippets, but the whole thing), thank you!
